Upgrade test-worker to use new Python for testing

jinchihe opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://www.kubeflow.org/docs/about/contributing/ and developer guide https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:

I updated the test-worker image that's used Python 3.8, and removed some code that upgrading Python from 3.5 to 3.6 in test script. Thanks.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #813

jinchihe commented:
Need clear up test env and retest later.

```
fatal: could not create leading directories of '/mnt/test-data-volume/kubeflow-kfserving-presubmit-e2e-819-7aa6c55-2176-41be/src/kubeflow
/kfserving': No space left on device
```

jlewi commented on test/scripts/pylint-test.sh:
```diff
@@ -0,0 +1,43 @@
+python3 -m pip install --upgrade pip
+pip3 install --upgrade pylint
+
+python -m kubeflow.testing.test_py_lint --artifacts_dir=$1 --src_dir=$2
```
See comment 
https://github.com/kubeflow/kfserving/issues/813#issuecomment-627324759

We already have a test worker image with python 3.8. So you could probably just use that.

jinchihe replied:
Thanks @jlewi , Will try use new image.

cliveseldon replied:
Did we not want to try this?

jinchihe commented:
/retest

jinchihe commented:
/assign @cliveseldon 

I updated the test-worker image that's used Python 3.8, and removed some code that upgrading Python from 3.5 to 3.6 in test script. Thanks.

cliveseldon commented:
/ok-to-test

cliveseldon commented:
/approve
/lgtm
